crates/cairo-profiler/src/trace_reader/function_trace_builder.rs

```diff
@@ -191,9 +158,7 @@ pub fn collect_function_level_profiling_info(
                 }
             }
             GenStatement::Return(_) => {
-                if call_stack.exit_function_call().is_none() {
-                    end_of_program_reached = true;
-                }
```

Comment:
This is still a valuable sanity check imo. After reaching end of the program you can only go into header, you can still check it

Reply:
Hmm, ok I will re-add this then. 
Maybe I will also add a test for it? 🤔 I have removed it since I thought it is redundant now, but indeed there might be a case someone changes things the wrong way and introduce a bug

Reply:
I have re-added this in https://github.com/software-mansion/cairo-profiler/pull/122/commits/1ec7a8c719f59800b1f2afac66c40cfb217705f4

Also updated task to add tests with this thread, to make sure such cases are properly tested https://github.com/software-mansion/cairo-profiler/issues/87